### PR TITLE
Release test items

### PR DESCRIPTION
**Pull request by nickhsine**

### CODE REVIEW

File: release-test-items/users.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+3. 在第一個輸入框中輸入`測試用帳號`
+4. 在 `Email`輸入框中輸入`test@twreporter.org`
+5. 在 `Password` 輸入框中連續輸入 `1234`
+6. 按下 Create 按鈕
+7. 畫面會出現 `New User 測試用帳號 created`
+
+---
+
+#### 刪除 User
+1. 返回 Users 列表頁 (`/keystone/users`)
```
Comment (babygoat) [inline release-test-items/users.md:66]: 建議testing items間不要有order dependencies, 這邊改成完成 新增User

File: release-test-items/users.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+### go to  `/keystone/users` 
+
+#### 確定 Users 數量：
+Users 數量必須與 mongo 資料庫中的 `users collection`  的數量相同。
+
+---
+
+#### 測試 Search 功能：
+在 Search 的輸入框中，輸入文字`林` ，畫面中的 Users 列表要剩下有文字`林`的 Users。
```
Comment (babygoat) [inline release-test-items/users.md:11]: 建議同步下面格式
1.  在 Search 的輸入框中，輸入文字`林`
2. 確認畫面中的 Users 列表剩下有文字`林`的 Users

File: release-test-items/users.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+## 測試 Users 頁面
+
+### go to  `/keystone/users` 
+
+#### 確定 Users 數量：
+Users 數量必須與 mongo 資料庫中的 `users collection`  的數量相同。
```
Comment (babygoat) [inline release-test-items/users.md:6]: 這部分我想了解這個item的測試目的？是測試有沒有正確連線？還是測試db.users.count()的正確性？

另外建議同步下面格式

Comment (nickhsine) [reply release-test-items/users.md:6]: 加入這個測試的原因是：
上次我在 local 端 restore mongo database，雖然有 index invalid 的問題，但 restore 還是有執行完成。
當時，我用 local keystone 檢查時，發現幾個頁面的數量都是 0。
所以我特地加了這個測試條件。

Comment (babygoat) [reply release-test-items/users.md:6]: test item的目標是確認軟體運行是否正常（如同功能描述）且每次都需要測試，我認為是不需要每次都精確到"數量"本身的，而且這樣的UI測試ground truth取得並不容易，所以建議是拿掉

你提到這個部分其實是collection metadata的問題，是mongorestore完後的檢查，要per collection跑一次validation`db.collection.validate(true)`確認狀態是否正常的restore檢驗，應該不屬於keystone範疇

**Comment by nickhsine** [comment]
Updated. Ready to review.

File: release-test-items/posts.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+1. 根據「[更新 Post](#更新-post)」，產生測試文章
+2. go to https://keystone-preview.twreporter.org/a/test-slug
+3. 檢查頁面上的資訊是否正確
+
+---
+
+#### 檢查聊天室
+1. 根據「[更新 Post](#更新-post)」，產生測試文章
+2. 畫面會停留在該篇新增文章的頁面
+3. 確認畫面右方是否出現聊天室，見範例圖（）
```
Comment (taylrj) [inline release-test-items/posts.md:158]: 範例圖？

**Review by taylrj — APPROVED** [review]
LGTM 👍

**Comment by babygoat** [comment]
I'm unable to start the keystone with the latest commits. Am I miss something?

```
ReferenceError: Unknown keystone list "companies"
    at Keystone.list (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/list.js:12:21)
    at /Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/initNav.js:42:25
    at Function._.map._.collect (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/node_modules/underscore/underscore.js:172:24)
    at /Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/initNav.js:40:21
    at Function._.each._.forEach (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/node_modules/underscore/underscore.js:158:9)
    at Keystone.initNav (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/initNav.js:31:4)
    at Keystone.exports.set (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/options.js:67:21)
    at Object.<anonymous> (/Users/babygoat/workarea/plate/keystone.js:9:10)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

**Comment by nickhsine** [comment]
> I'm unable to start the keystone on environment-setup with the latest commits. Am I miss something?
> 
> ```
> ReferenceError: Unknown keystone list "companies"
>     at Keystone.list (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/list.js:12:21)
>     at /Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/initNav.js:42:25
>     at Function._.map._.collect (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/node_modules/underscore/underscore.js:172:24)
>     at /Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/initNav.js:40:21
>     at Function._.each._.forEach (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/node_modules/underscore/underscore.js:158:9)
>     at Keystone.initNav (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/initNav.js:31:4)
>     at Keystone.exports.set (/Users/babygoat/workarea/plate/node_modules/@twreporter/keystone/lib/core/options.js:67:21)
>     at Object.<anonymous> (/Users/babygoat/workarea/plate/keystone.js:9:10)
>     at Module._compile (internal/modules/cjs/loader.js:778:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
>     at Module.load (internal/modules/cjs/loader.js:653:32)
>     at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
>     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
>     at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
>     at startup (internal/bootstrap/node.js:283:19)
>     at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
> ```

Could you help to check `plate/models/` folder ?
`companies`, `image-collection` ... etc should be removed.

**Comment by babygoat** [comment]
> Could you help to check `plate/models/` folder ?
> `company`, `image-collection` ... etc should be removed in [this commit](https://github.com/twreporter/plate/commit/6269b0c4a602166451aad9d032cd98c43069e9cb#diff-8edcc26c0f6c7256a71a8ede6b954ad9ec8dabf88f127c5e9d34f13cadefe11a).
> 

Thanks. The issue results from model configuration in the config.js which requires clean build.
```
exports.nav = {
  'people': ['users', 'contacts'],
  'content': ['posts', 'post-categories', 'topics', 'images', 'audios', 'videos', 'tags'],
};
```

> Maybe try to rebuild `config.js` by `make build-conifg`.

File: release-test-items/contacts.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+1. 點選`職稱`
+2. 選擇`Matches`選項
+3. 選擇`Contains`選項
+4. 在輸入框中，輸入文字 `特約記者`
+5. 按下 Apply 按鈕
+6. 列表會被過濾，只秀出`職稱`中包含`特約記者`字串的資料列
+
+##### 列表 filtered by `Email`
+1. 點選`Email`
+2. 選擇 `Matches` 選項
```
Comment (babygoat) [inline release-test-items/contacts.md:45]: tab(分頁）？

File: release-test-items/contacts.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
+## 測試 Contacts 頁面
+
+### go to  `/keystone/contacts` 
+
+---
+
+#### 測試 Search 功能：
+1. 在 Search 的輸入框中，輸入文字`林` 
+2. 確認畫面中的 Users 列表要剩下有文字`林`的 Users。
```
Comment (babygoat) [inline release-test-items/contacts.md:9]: Contacts列表 or  Users?

File: release-test-items/contacts.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
+## 測試 Contacts 頁面
+
+### go to  `/keystone/contacts` 
+
+---
+
+#### 測試 Search 功能：
```
Comment (babygoat) [inline release-test-items/contacts.md:7]: Search(作者姓名）

File: release-test-items/contacts.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+4. 在 `Email`輸入框中輸入`test@twreporter.org`
+6. 按下 Create 按鈕
+7. 畫面會出現 `New Contact 測試用帳號 created`
+
+---
+
+#### 刪除 Contact
+1. 根據「[新增 Contact](#新增-contact)」，產生欲刪除的帳號
+2. 到 Contacts 列表頁 (`/keystone/contacts`)
+3. 找到 Name 是 `測試用帳號` 的資料列
```
Comment (babygoat) [inline release-test-items/contacts.md:66]: 作者姓名？

File: release-test-items/contacts.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+4. 點選資料列前的垃圾桶 icon
+5. 點選 Delete 按鈕
+6. 確認列表中，沒有 `測試用帳號` 資料列
+
+---
```
Comment (babygoat) [inline release-test-items/contacts.md:71]: 確認一下，這邊名詞偶有混用狀況是因為包含 `Contacts` 及 `Users` 分頁？

Comment (babygoat) [reply release-test-items/contacts.md:71]: What about 編輯聯絡人？想確認這邊是不是硬需求？

File: release-test-items/images.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+#### 刪除 Image
+1. 根據「[新增 Image](#新增-image)」，產生測試照片
+2. 根據「[測試 Search 功能](#測試-search-功能)」，找到測試照片
+3. 點選資料列前的垃圾桶 icon
+4. 點選`Delete`按鈕
+5. 確認列表中，沒有`Description`是`test-image`的資料列
+
+---
```
Comment (babygoat) [inline release-test-items/images.md:30]: What about 編輯image？(keyword? copyright?) 想確認這邊是不是硬需求？

File: release-test-items/posts.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
+## 測試 Posts 頁面
+
+### go to  `/keystone/posts` 
+
+---
+
+#### 測試 Search 功能：
```
Comment (babygoat) [inline release-test-items/posts.md:7]: Search  網址名稱（英文）

File: release-test-items/posts.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+2. 標題: 輸入`測試文章標題`
+3. 副標：輸入`測試文章副標題`
+4. 狀態: 維持`Draft`
+5. 作者：選擇`柯皓翔`
+6. 攝影：選擇`余志偉`
+7. 設計：選擇`林珍娜`
+8. 工程：選擇`方泰鈞`
+9. 作者（其他）： 輸入`共同採訪：王儀君`
+10. 首圖：點選`Select Photo`，選擇第一張照片
+11. 首圖：點選`Select Photo`，選擇第二張照片
```
Comment (babygoat) [inline release-test-items/posts.md:124]: 首圖portraint ?

File: release-test-items/posts.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+2. go to https://keystone-preview.twreporter.org/a/test-slug
+3. 檢查頁面上的資訊是否正確
+
+---
+
+#### 檢查聊天室
+1. 根據「[更新 Post](#更新-post)」，產生測試文章
+2. 畫面會停留在該篇新增文章的頁面
+3. 確認畫面右方是否出現聊天室，見範例圖（）
+
```
Comment (babygoat) [inline release-test-items/posts.md:159]: 需要測試算字數功能？

File: release-test-items/topics.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
+## 測試 Topics 頁面
+
+### go to  `/keystone/topics` 
+
+---
+
+#### 測試 Search 功能：
```
Comment (babygoat) [inline release-test-items/topics.md:7]: Search 專題名稱

File: release-test-items/topics.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+13. `首圖`
+14. `首圖（Portrait）`
+15. `置頂影片`
+16. `分享標題`
+17. `分享說明`
+18. `分享縮圖`
+
+其中 `專題名稱`和`發佈日期`預設是打勾的狀態，而其他則是尚未打勾的狀態。
+
+##### 列表裡列出`作者`欄位
```
Comment (babygoat) [inline release-test-items/topics.md:36]: 標題？

File: release-test-items/topics.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+11. `相關文章版型`
+12. `文章區背景色`
+13. `首圖`
+14. `首圖（Portrait）`
+15. `置頂影片`
+16. `分享標題`
+17. `分享說明`
+18. `分享縮圖`
+
+##### 列表 filtered by `標題`
```
Comment (babygoat) [inline release-test-items/topics.md:65]: 專題名稱？

File: release-test-items/topics.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+5. 按下 Apply 按鈕
+6. 列表會被過濾，只秀出`專題名稱`中包含`香港`字串的資料列
+
+##### 列表 filtered by `發佈日期`
+1. 點選`發佈日期`
+2. 選擇 `Matches` 選項
+3. 選擇 `After` 選項
+4. 在日期表裡選擇欲查詢的發佈日期。舉例而言，在日期表裡挑選2020年7月1日 
+5. 按下 Apply 按鈕
+6. 列表會被過濾，只秀出`發佈日期`在`2020-07-01`**之後**的文章
```
Comment (babygoat) [inline release-test-items/topics.md:79]: （不包含）

File: release-test-items/users.md [hunk 18]
```diff
@@ -0,0 +1,71 @@
+## 測試 Users 頁面
+
+### go to  `/keystone/users` 
+
+---
+
+#### 測試 Search 功能：
```
Comment (babygoat) [inline release-test-items/users.md:7]: Search name

**Comment by babygoat** [comment]
close & defer the further improvement to [the story](https://kanbanflow.com/t/jJ9EkR2c)

**Pull request merged**